src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -203,6 +203,9 @@ CHIP_ERROR AdvertiserMinMdns::Start(chip::Inet::InetLayer * inetLayer, uint16_t
 
     mCommissionInstanceName1 = GetRandU32();
     mCommissionInstanceName2 = GetRandU32();
+    // Re-set the server in the response sender in case this has been swapped in the
+    // GlobalMinimalMdnsServer (used for testing).
+    mResponseSender.SetServer(&GlobalMinimalMdnsServer::Server());
```

Comment:
This feels off.

Could we force the swap to be always scoped, like having a code of:

```cpp
{
   ScopedServerReplace replace(GlobalMinimalMdnsServer::Instance(), &server);
   ///  .... server is replaced in the block
 } 
 /// server stops being replaced
 ```
 
 and enforce only this scoped replacement can do the get/set of the server?

Reply:
Misread  the code - thought we replace the reponser server, but rather  we replace the responder and need to tie it to the current server. Resolving.